crates/typst/src/layout/inline/shaping.rs

```diff
@@ -563,19 +563,6 @@ impl<'a> ShapedText<'a> {
             }
         };
 
-        let next = match towards {
-            Side::Left => usize::checked_sub,
-            Side::Right => usize::checked_add,
-        };
-
-        // Search for the outermost glyph with the text index.
-        while let Some(next) = next(idx, 1) {
-            if self.glyphs.get(next).map_or(true, |g| g.range.start != text_index) {
-                break;
-            }
-            idx = next;
-        }
```

Comment:
@laurmaedje Can you help me understand the scenarios where this deleted code would be used? Git blame seems not helpful. Based on my results from running all tests, this code is only triggered and produces incorrect output when encountering Thai language.